File: Day067_2025/Q067_GFG.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        if n == 0:
            return ""

        start, max_len = 0, 1

        for i in range(n):
            for j in range(2):  # j=0 for odd-length, j=1 for even-length
                low, high = i, i + j
                while low >= 0 and high < n and s[low] == s[high]:
                    curr_len = high - low + 1
                    if curr_len > max_len:
                        start = low
                        max_len = curr_len
                    low -= 1
                    high += 1

        return s[start:start + max_len]

File: Day067_2025/Q067_GFG.java
class Q067_GFG {
    public String longestPalindrome(String s) {
        int n = s.length();
        if (n == 0) return "";

        int start = 0, maxLen = 1;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j <= 1; j++) { // j=0 for odd-length, j=1 for even-length
                int low = i, high = i + j;
                while (low >= 0 && high < n && s.charAt(low) == s.charAt(high)) {
                    int currLen = high - low + 1;
                    if (currLen > maxLen) {
                        start = low;
                        maxLen = currLen;
                    }
                    low--;
                    high++;
                }
            }
        }

        return s.substring(start, start + maxLen);
    }
}
